File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/EquityStrike.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class EquityStrike : ISerialized
    {
        public EquityStrike(XmlNode xmlNode)
        {
            XmlNode strikePriceNode = xmlNode.SelectSingleNode("strikePrice");
            
            if (strikePriceNode != null)
            {
                if (strikePriceNode.Attributes["href"] != null || strikePriceNode.Attributes["id"] != null) 
                {
                    if (strikePriceNode.Attributes["id"] != null) 
                    {
                        strikePriceIDRef_ = strikePriceNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(strikePriceNode);
                        IDManager.SetID(strikePriceIDRef_, ob);
                    }
                    else if (strikePriceNode.Attributes["href"] != null)
                    {
                        strikePriceIDRef_ = strikePriceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        strikePrice_ = new XsdTypeDecimal(strikePriceNode);
                    }
                }
                else
                {
                    strikePrice_ = new XsdTypeDecimal(strikePriceNode);
                }
            }
            
        
            XmlNode strikePercentageNode = xmlNode.SelectSingleNode("strikePercentage");
            
            if (strikePercentageNode != null)
            {
                if (strikePercentageNode.Attributes["href"] != null || strikePercentageNode.Attributes["id"] != null) 
                {
                    if (strikePercentageNode.Attributes["id"] != null) 
                    {
                        strikePercentageIDRef_ = strikePercentageNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(strikePercentageNode);
                        IDManager.SetID(strikePercentageIDRef_, ob);
                    }
                    else if (strikePercentageNode.Attributes["href"] != null)
                    {
                        strikePercentageIDRef_ = strikePercentageNode.Attributes["href"].Value;
                    }
                    else
                    {
                        strikePercentage_ = new XsdTypeDecimal(strikePercentageNode);
                    }
                }
                else
                {
                    strikePercentage_ = new XsdTypeDecimal(strikePercentageNode);
                }
            }
            
        
            XmlNode strikeDeterminationDateNode = xmlNode.SelectSingleNode("strikeDeterminationDate");
            
            if (strikeDeterminationDateNode != null)
            {
                if (strikeDeterminationDateNode.Attributes["href"] != null || strikeDeterminationDateNode.Attributes["id"] != null) 
                {
                    if (strikeDeterminationDateNode.Attributes["id"] != null) 
                    {
                        strikeDeterminationDateIDRef_ = strikeDeterminationDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(strikeDeterminationDateNode);
                        IDManager.SetID(strikeDeterminationDateIDRef_, ob);
                    }
                    else if (strikeDeterminationDateNode.Attributes["href"] != null)
                    {
                        strikeDeterminationDateIDRef_ = strikeDeterminationDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        strikeDeterminationDate_ = new AdjustableOrRelativeDate(strikeDeterminationDateNode);
                    }
                }
                else
                {
                    strikeDeterminationDate_ = new AdjustableOrRelativeDate(strikeDeterminationDateNode);
                }
            }
            
        
            XmlNode currencyNode = xmlNode.SelectSingleNode("currency");
            
            if (currencyNode != null)
            {
                if (currencyNode.Attributes["href"] != null || currencyNode.Attributes["id"] != null) 
                {
                    if (currencyNode.Attributes["id"] != null) 
                    {
                        currencyIDRef_ = currencyNode.Attributes["id"].Value;
                        Currency ob = new Currency(currencyNode);
                        IDManager.SetID(currencyIDRef_, ob);
                    }
                    else if (currencyNode.Attributes["href"] != null)
                    {
                        currencyIDRef_ = currencyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        currency_ = new Currency(currencyNode);
                    }
                }
                else
                {
                    currency_ = new Currency(currencyNode);
                }
            }
            
        
        }
        
    
        #region StrikePrice_
        private XsdTypeDecimal strikePrice_;
        public XsdTypeDecimal StrikePrice_
        {
            get
            {
                if (this.strikePrice_ != null)
                {
                    return this.strikePrice_; 
                }
                else if (this.strikePriceIDRef_ != null)
                {
                    strikePrice_ = IDManager.getID(strikePriceIDRef_) as XsdTypeDecimal;
                    return this.strikePrice_; 
                }
                else
                {
                      return this.strikePrice_; 
                }
            }
            set
            {
                if (this.strikePrice_ != value)
                {
                    this.strikePrice_ = value;
                }
            }
        }
        #endregion
        
        public string strikePriceIDRef_ { get; set; }
        #region StrikePercentage_
        private XsdTypeDecimal strikePercentage_;
        public XsdTypeDecimal StrikePercentage_
        {
            get
            {
                if (this.strikePercentage_ != null)
                {
                    return this.strikePercentage_; 
                }
                else if (this.strikePercentageIDRef_ != null)
                {
                    strikePercentage_ = IDManager.getID(strikePercentageIDRef_) as XsdTypeDecimal;
                    return this.strikePercentage_; 
                }
                else
                {
                      return this.strikePercentage_; 
                }
            }
            set
            {
                if (this.strikePercentage_ != value)
                {
                    this.strikePercentage_ = value;
                }
            }
        }
        #endregion
        
        public string strikePercentageIDRef_ { get; set; }
        #region StrikeDeterminationDate_
        private AdjustableOrRelativeDate strikeDeterminationDate_;
        public AdjustableOrRelativeDate StrikeDeterminationDate_
        {
            get
            {
                if (this.strikeDeterminationDate_ != null)
                {
                    return this.strikeDeterminationDate_; 
                }
                else if (this.strikeDeterminationDateIDRef_ != null)
                {
                    strikeDeterminationDate_ = IDManager.getID(strikeDeterminationDateIDRef_) as AdjustableOrRelativeDate;
                    return this.strikeDeterminationDate_; 
                }
                else
                {
                      return this.strikeDeterminationDate_; 
                }
            }
            set
            {
                if (this.strikeDeterminationDate_ != value)
                {
                    this.strikeDeterminationDate_ = value;
                }
            }
        }
        #endregion
        
        public string strikeDeterminationDateIDRef_ { get; set; }
        #region Currency_
        private Currency currency_;
        public Currency Currency_
        {
            get
            {
                if (this.currency_ != null)
                {
                    return this.currency_; 
                }
                else if (this.currencyIDRef_ != null)
                {
                    currency_ = IDManager.getID(currencyIDRef_) as Currency;
                    return this.currency_; 
                }
                else
                {
                      return this.currency_; 
                }
            }
            set
            {
                if (this.currency_ != value)
                {
                    this.currency_ = value;
                }
            }
        }
        #endregion
        
        public string currencyIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/EquityStrike.cpp
// EquityStrike.cpp 
#include "EquityStrike.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

EquityStrike::EquityStrike(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //strikePriceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* strikePriceNode = xmlNode->FirstChildElement("strikePrice");

   if(strikePriceNode){strikePriceIsNull_ = false;}
   else{strikePriceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- strikePriceNode , address : " << strikePriceNode << std::endl;
   #endif
   if(strikePriceNode)
   {
      if(strikePriceNode->Attribute("href") || strikePriceNode->Attribute("id"))
      {
          if(strikePriceNode->Attribute("id"))
          {
             strikePriceIDRef_ = strikePriceNode->Attribute("id");
             strikePrice_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(strikePriceNode));
             strikePrice_->setID(strikePriceIDRef_);
             IDManager::instance().setID(strikePriceIDRef_,strikePrice_);
          }
          else if(strikePriceNode->Attribute("href")) { strikePriceIDRef_ = strikePriceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { strikePrice_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(strikePriceNode));}
   }

   //strikePercentageNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* strikePercentageNode = xmlNode->FirstChildElement("strikePercentage");

   if(strikePercentageNode){strikePercentageIsNull_ = false;}
   else{strikePercentageIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- strikePercentageNode , address : " << strikePercentageNode << std::endl;
   #endif
   if(strikePercentageNode)
   {
      if(strikePercentageNode->Attribute("href") || strikePercentageNode->Attribute("id"))
      {
          if(strikePercentageNode->Attribute("id"))
          {
             strikePercentageIDRef_ = strikePercentageNode->Attribute("id");
             strikePercentage_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(strikePercentageNode));
             strikePercentage_->setID(strikePercentageIDRef_);
             IDManager::instance().setID(strikePercentageIDRef_,strikePercentage_);
          }
          else if(strikePercentageNode->Attribute("href")) { strikePercentageIDRef_ = strikePercentageNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { strikePercentage_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(strikePercentageNode));}
   }

   //strikeDeterminationDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* strikeDeterminationDateNode = xmlNode->FirstChildElement("strikeDeterminationDate");

   if(strikeDeterminationDateNode){strikeDeterminationDateIsNull_ = false;}
   else{strikeDeterminationDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- strikeDeterminationDateNode , address : " << strikeDeterminationDateNode << std::endl;
   #endif
   if(strikeDeterminationDateNode)
   {
      if(strikeDeterminationDateNode->Attribute("href") || strikeDeterminationDateNode->Attribute("id"))
      {
          if(strikeDeterminationDateNode->Attribute("id"))
          {
             strikeDeterminationDateIDRef_ = strikeDeterminationDateNode->Attribute("id");
             strikeDeterminationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(strikeDeterminationDateNode));
             strikeDeterminationDate_->setID(strikeDeterminationDateIDRef_);
             IDManager::instance().setID(strikeDeterminationDateIDRef_,strikeDeterminationDate_);
          }
          else if(strikeDeterminationDateNode->Attribute("href")) { strikeDeterminationDateIDRef_ = strikeDeterminationDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { strikeDeterminationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(strikeDeterminationDateNode));}
   }

   //currencyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* currencyNode = xmlNode->FirstChildElement("currency");

   if(currencyNode){currencyIsNull_ = false;}
   else{currencyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- currencyNode , address : " << currencyNode << std::endl;
   #endif
   if(currencyNode)
   {
      if(currencyNode->Attribute("href") || currencyNode->Attribute("id"))
      {
          if(currencyNode->Attribute("id"))
          {
             currencyIDRef_ = currencyNode->Attribute("id");
             currency_ = boost::shared_ptr<Currency>(new Currency(currencyNode));
             currency_->setID(currencyIDRef_);
             IDManager::instance().setID(currencyIDRef_,currency_);
          }
          else if(currencyNode->Attribute("href")) { currencyIDRef_ = currencyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { currency_ = boost::shared_ptr<Currency>(new Currency(currencyNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDecimal> EquityStrike::getStrikePrice()
{
   if(!this->strikePriceIsNull_){
        if(strikePriceIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(strikePriceIDRef_));
        }else{
             return this->strikePrice_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<XsdTypeDecimal> EquityStrike::getStrikePercentage()
{
   if(!this->strikePercentageIsNull_){
        if(strikePercentageIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(strikePercentageIDRef_));
        }else{
             return this->strikePercentage_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<AdjustableOrRelativeDate> EquityStrike::getStrikeDeterminationDate()
{
   if(!this->strikeDeterminationDateIsNull_){
        if(strikeDeterminationDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(strikeDeterminationDateIDRef_));
        }else{
             return this->strikeDeterminationDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
boost::shared_ptr<Currency> EquityStrike::getCurrency()
{
   if(!this->currencyIsNull_){
        if(currencyIDRef_ != ""){
             return boost::shared_static_cast<Currency>(IDManager::instance().getID(currencyIDRef_));
        }else{
             return this->currency_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Currency>();
   }
}
}

